### ISSUE TITLE
Add transparent background option

### ISSUE THREAD

**Issue by xaolin** [lay295/TwitchDownloader#16]
title

**Comment by xaolin** [comment]
Or you could add an opacity option

**Comment by lay295** [comment]
You can give the new version a try. Works well for me at the moment :)

You can either do .mov and RLE codec or .webm with VP8/VP9

Do note that if you go with .mov, the resulting file sizes will he huge. Probably 100x the size with H264 compression. With webm it'll be a good manageable file size, but I'm unsure if video editors support webm. I could not find any other format/codec combination that ffmpeg can encode with alpha support.

**Comment by xaolin** [comment]
> You can give the new version a try. Works well for me at the moment :)
> 
> You can either do .mov and RLE codec or .webm with VP8/VP9
> 
> Do note that if you go with .mov, the resulting file sizes will he huge. Probably 100x the size with H264 compression. With webm it'll be a good manageable file size, but I'm unsure if video editors support webm. I could not find any other format/codec combination that ffmpeg can encode with alpha support.

Hey dude. Thanks for the consistent updates. Tbh I didn't need those ones. What I needed were some more customization options for the chat. Mainly reducing background opacity or even adding feather to the edges or stroke.

**Comment by lay295** [comment]
Yeah you can set the background opacity to 0 now (transparent), again file size will be huge with .mov though. I haven't really tied any opacity besides 0% and 100% so it might not work well.

**Comment by iWannaSilence** [comment]
> You can either do .mov and RLE codec or .webm with VP8/VP9

Hello. I have some problems with RLE .mov files, Adobe Premiere Pro cannot import .mov with RLE codec (The File has no audio or video stream), but ProRes works fine. With transcoding with ffmpeg, it starts to work. Can you add Prores or something that supports Adobe Premiere with alpha?

.WEBM is rendered for a very long time compared to .MOV. If you download a chat lasting 1 hour and start rendering, then .MOV will take ~ 5 minutes maximum in time, when .WEBM is more than three hours. Moreover, I still have not checked the output file whether it really has a working alpha channel.

Yes, I know that there is another way to get a transparent chat from a non-working .MOV file by creating a sequence from .PNG files. But, as you already understood, 1 hour of video in 30 FPS in the form of .PNG sequences is a real pain.

**Comment by lay295** [comment]
I can add ProRes, I usually use Sony Vegas which worked with RLE. I really have no idea why VP8/9 take so long and don't have the time to look into it currently.

But yeah all the codecs that support alpha seem to have massive file sizes except like VP8/9 but that takes forever to encode it seems, I've tried to tweak it with various command line arguments but the best I could get is like 0.25x realtime which is pretty bad.

Alternatives would be to just use good ol H264 and mp4 and try and chroma key it out. 
If you don't wanna chroma key it out you either have to sacrifice file size or render time, I really haven't found any other better options unfortunately :/ I've looked for hours. Any suggestions are welcome. I can try to figure out DNxHD later but it seems picky with options. In the meantime if you haven't figured it out and want to use ProRes just when RLE is selected edit the command line options and change "-c:v qtrle" to "-c:v prores_ks"
